### PR TITLE
Eth-connector: change `total_supply` behaviour. Refactoring.

### PR DESCRIPTION
**Pull request by sept-en**
This is a follow-up to #133 PR. This is required as that PR contains needed changes so we won't fix tests twice just to make CI pass. Once that PR is merged, the current one needs to be rebased on top of `develop` branch.

Following our discussion with @mfornet, we need to change the `ft_total_supply` meaning to correspond to the NEP-141 standard. Also, some other refactoring was required as many members of the team frequently get confusing with `ft_total_supply_near` thinking that this represents bridged Near to Aurora (which we don't have) and other similar things.

Please note, that this PR most likely will require state migration as inner fields of the `FungibleToken` structure were renamed. However, this should be ok on Mainnet as we don't have an eth-connector available there.

Changes:
* `ft_total_supply()` now represents the amount of nETH (ETH NEP-141 tokens on NEAR) to correspond to NEP-141 standard.
* `FungibleToken`: rename `total_supply` field -> `total_eth_supply_on_near`.
* `FungibleToken`: rename `total_supply_eth` field -> `total_eth_supply_on_aurora`.
* Rename `ft_total_supply_near()` -> `ft_total_eth_supply_on_near()`.
* Rename `ft_total_supply_eth()` -> `ft_total_supply_eth_on_near()`.
* Rename `finish_deposit_near()` -> `finish_deposit()`.
* `deposit()` method refactoring.
* Minor refactoring.
* Fix tests according to changes and refactoring.

### CODE REVIEW

**Review by mfornet — COMMENTED** [review]
I appreciate this level of verbosity for function names. Using functions like `ft_total_supply_near` was not useful to understand what was happening there.

**Comment by artob** [comment]
> This is a follow-up to #133 PR. This is required as that PR contains needed changes so we won't fix tests twice just to make CI pass. Once that PR is merged, the current one needs to be rebased on top of `develop` branch.

@sept-en I've merged #133, so please proceed with the rebase here.

File: src/fungible_token.rs [hunk 1]
```diff
@@ -16,11 +16,11 @@ const GAS_FOR_FT_ON_TRANSFER: Gas = 10_000_000_000_000;
...
 #[derive(Debug, Default, BorshDeserialize, BorshSerialize)]
 pub struct FungibleToken {
-    /// Total supply of the all token.
-    pub total_supply: Balance,
+    /// Total ETH supply of ETH on Near (nETH as NEP-141 token)
+    pub total_eth_supply_on_near: Balance,
 
-    /// Total supply of the all ETH token.
-    pub total_supply_eth: Balance,
+    /// Total ETH supply on Aurora (native ETH in Aurora EVM)
```
[the remark this answers is not in the document]
Comment (sept-en) [reply src/fungible_token.rs:22]: The idea was to emphasize that this is the native currency in Aurora EVM. nETH (NEP-141) is non-native.
But I'll remove it as it looks confusing to you!

**Comment by sept-en** [comment]
@artob rebase is finished.

**Comment by sept-en** [comment]
Update:
@mfornet pointed out that as `FungibleToken` is being borsh-serialized when we save it to storage, it seems that we should fine and don't need any storage migration (as we neither added any new fields nor changed the order of variables in the current PR).

**Review by birchmd — APPROVED** [review]
LGTM! Thanks for this!

**Comment by mrLSD** [comment]
@sept-en @mfornet 
I think this name is unfortunate:
* `*_on_near`
* `*_on_aurora`

But in reality it means:
* `*_for_near_account`
* `*_for_eth_account`

For me, this naming is misleading.

However, if it blocks somehow release, I will do approve immediately. Just let me know.

**Comment by sept-en** [comment]
@mrLSD `for_eth_account` is not really concise suffix because it's not really clear whether it's about Aurora or Ethereum eth account.

If we look at the Near and Aurora as separate entities (not taking into account that Aurora EVM is inside the Near account), we can say that there are no Aurora accounts on Near and vice versa.

So I think the naming is quite clear if we get used to it.
Moreover, we already use similar naming in `EthCustodian` for several months already.

**Comment by mrLSD** [comment]
> If we look at the Near and Aurora as separate entities (not taking into account that Aurora EVM is inside the Near account), we can say that there are no Aurora accounts on Near and vice versa.

> If we look at the Near and Aurora as separate entities (not taking into account that Aurora EVM is inside the Near account), we can say that there are no Aurora accounts on Near and vice versa.

For me, it's not clear how it affects current accounts balance logic.
1. We deposit via Proof to Aurora Near accounts
1. We deposit via Proof to Aurora Eth accounts
3. Aurora doesn't have its own accounts and account management itself. Taking this into account, the proposed naming for me does not reflect this logic. 

> So I think the naming is quite clear if we get used to it.
> Moreover, we already use similar naming in `EthCustodian` for several months already.

It doesn't mean that if we already use that naming anywhere that naming is clear.

**Review by joshuajbouw — APPROVED** [review]
Straight forward. Thanks for this.

**Review by mfornet — APPROVED** [review]
Great, much easier to use this names.

**Review by artob — APPROVED** [review]

**Comment by artob** [comment]
@sept-en Please fix the merge conflict (due to #51) and then I will merge.

**Comment by sept-en** [comment]
> Aurora doesn't have its own accounts and account management itself. Taking this into account, the proposed naming for me does not reflect this logic.

The idea for these names is to represent these balances from a top-level conceptual point of view. We don't want to have very detailed implementation specifics represented in function names. I think having top-level abstractions is a good approach.

**Comment by sept-en** [comment]
@artob merged. Please check!

**Pull request merged**